site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -277,6 +277,34 @@ public boolean filterUserSites(List<SiteDTO> sitesReturnObject,
 
     }
 
+    @Override
+    public List<SiteNameDTO> findAssignedSites() {
+        //get unfiltered roleAssignments
+        List<RoleAssignmentDTO> roleAssnmts = new ArrayList<>(
+            practServClnt.getUserRoleAssignments(authService.getUserId(), authService.getAuthHeaders()));
+
+        final List<SiteDTO> sites = this.siteStore.findAll().stream()
+            .map(site -> convertSite(site))
+            .collect(Collectors.toList());
+
+        if (sites.isEmpty()) {
+            throw new InvariantException(Site.NO_ROOT_SITE.message());
+        }
+
+        if (roleAssnmts.isEmpty()) {
```

Comment:
Is it worth moving this check to earlier in the flow - before the sites retrieval - since if we are going to throw this error then we don't need to bother retrieving sites?

Reply:
oops I had moved it but didnt commit..